src/vs/workbench/contrib/inlineChat/browser/inlineChatDecorations.ts

```diff
@@ -0,0 +1,161 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Codicon } from 'vs/base/common/codicons';
+import { ThemeIcon } from 'vs/base/common/themables';
+import { ICodeEditor, IEditorMouseEvent } from 'vs/editor/browser/editorBrowser';
+import { IEditorContribution } from 'vs/editor/common/editorCommon';
+import { GlyphMarginLane, IModelDecorationsChangeAccessor, ITextModel, TrackedRangeStickiness } from 'vs/editor/common/model';
+import { ModelDecorationOptions } from 'vs/editor/common/model/textModel';
+import { localize } from 'vs/nls';
+import { registerIcon } from 'vs/platform/theme/common/iconRegistry';
+import { Selection } from 'vs/editor/common/core/selection';
+import { InlineChatController } from 'vs/workbench/contrib/inlineChat/browser/inlineChatController';
+import { IConfigurationService, IConfigurationChangeEvent } from 'vs/platform/configuration/common/configuration';
+import { DisposableStore, Disposable } from 'vs/base/common/lifecycle';
+import { GutterActionsRegistry } from 'vs/workbench/contrib/codeEditor/browser/editorLineNumberMenu';
+import { Action } from 'vs/base/common/actions';
+import { CTX_INLINE_CHAT_VISIBLE, IInlineChatService } from 'vs/workbench/contrib/inlineChat/common/inlineChat';
+import { RunOnceScheduler } from 'vs/base/common/async';
+import { IContextKeyService } from 'vs/platform/contextkey/common/contextkey';
+import { Iterable } from 'vs/base/common/iterator';
+
+const gutterInlineChatIcon = registerIcon('inline-chat', Codicon.sparkle, localize('startInlineChatIcon', 'Icon which spawns the inline chat from the gutter'));
+
+export class InlineChatDecorationsContribution extends Disposable implements IEditorContribution {
+
+	private localToDispose = new DisposableStore();
```

Comment:
I suggest to prefix all private fields with `_`. That's how it's done at most places.
Status Const should be capitalized (gutterSettingID, gutterIconClassName)